### PR TITLE
God

### PR DESCRIPTION
**Pull request by shashvatshahi1998**
#27044

### CODE REVIEW

**Comment by shashvatshahi1998** [comment]
Guys please help me I am tired of these merged conflicts whenever I am adding a new commit all my past commits are automatically get  added in my pull request what to do????

**Comment by gshashank84** [comment]
> Guys please help me I am tired of these merged conflicts whenever I am adding a new commit all my past commits are automatically get added in my pull request what to do????

You can upstream the forked repo and merge with the new commits you have done.

File: tensorflow/python/eager/backprop.py [hunk 1]
```diff
@@ -209,8 +209,12 @@ def grad_fn(*args, **kwds):
     if not variables:
       raise ValueError("No trainable variables were accessed while the "
                        "function was being computed.")
-
+    for v in variables:  
+      if is_floating(v) == False:
```
Comment (alextp) [inline tensorflow/python/eager/backprop.py]: "if not is_floating(v)" or the linter will fail

File: tensorflow/python/eager/backprop.py [hunk 2]
```diff
@@ -209,8 +209,12 @@ def grad_fn(*args, **kwds):
...
+        raise ValueError("Gradient calculation supports only float types.")
```
Comment (alextp) [inline tensorflow/python/eager/backprop.py]: Print more information about the non-float variable being watched.

File: tensorflow/python/ops/string_ops.py [hunk 3]
```diff
@@ -193,7 +193,7 @@ def string_format(template, inputs, placeholder="{}", summarize=3, name=None):
 
 
 @tf_export("string_split")
-def string_split(source, delimiter=" ", skip_empty=True):  # pylint: disable=invalid-name
+def string_split(source, delimiter=" ", skip_empty=True, maxsplit=-1):  # pylint: disable=invalid-name
```
Comment (alextp) [inline tensorflow/python/ops/string_ops.py]: Unrelated changes?

File: tensorflow/python/ops/string_ops.py [hunk 4]
```diff
@@ -243,7 +243,7 @@ def string_split(source, delimiter=" ", skip_empty=True):  # pylint: disable=inv
 
 
 @tf_export("strings.split")
-def string_split_v2(source, sep=None, maxsplit=-1):
+def string_split_v2(source, sep=None, skip_empty=True,  maxsplit=-1):
```
Comment (alextp) [inline tensorflow/python/ops/string_ops.py]: Unrelated changes?

**Review by alextp — CHANGES_REQUESTED** [review]

File: tensorflow/python/eager/backprop.py (hunk 1 above)
Comment (superbobry) [reply tensorflow/python/eager/backprop.py]: This check probably belongs in `GradientTape.watch` before we even get to backprop. Wdyt?

File: tensorflow/core/ops/math_ops.cc [hunk 5]
```diff
@@ -460,7 +460,7 @@ REGISTER_OP("DivNoNan")
     .Input("x: T")
     .Input("y: T")
     .Output("z: T")
-    .Attr("T: {float, double}")
+    .Attr("T: {float, double, bfloat16, half}")
```
Comment (superbobry) [inline tensorflow/core/ops/math_ops.cc]: This is unrelated, right?

**Comment by rthadur** [comment]
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!
